libcodechecker/analyze/log_parser.py

```diff
@@ -26,14 +27,90 @@
     ['^-m(32|64)',
      '^-std=.*'])
 
+compiler_includes_dump_file = "compiler_includes.json"
+compiler_target_dump_file = "compiler_target.json"
+
 
-def get_compiler_includes(compiler, lang, compile_opts, extra_opts=None):
+def get_compiler_err(cmd):
     """
-    Returns a list of default includes of the given compiler.
+    Returns the stderr of a compiler invocation as string.
     """
+    try:
+        proc = subprocess.Popen(shlex.split(cmd),
+                                stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+
+        _, err = proc.communicate("")
+        return err
+    except OSError as oerr:
+        LOG.error("Error during process execution: " + cmd + '\n' +
+                  oerr.strerror + "\n")
+
+
+def parse_compiler_includes(lines):
+    '''
+    Parse the compiler include paths from a string
+    '''
     start_mark = "#include <...> search starts here:"
     end_mark = "End of search list."
 
+    include_paths = []
+
+    do_append = False
+    for line in lines.splitlines(True):
+        line = line.strip()
+        if line.startswith(end_mark):
+            break
+        if do_append:
+            # On OSX there are framework includes,
+            # where we need to strip the "(framework directory)" string.
+            # For instance:
+            # /System/Library/Frameworks (framework directory)
+            fpos = line.find("(framework directory)")
+            if fpos == -1:
+                include_paths.append("-isystem " + line)
+            else:
+                include_paths.append("-isystem " + line[0:fpos-1])
+
+        if line.startswith(start_mark):
+            do_append = True
+
+    return include_paths
+
+
+def parse_compiler_target(lines):
+    """
+    Parse the compiler target from a string.
+
+    """
+    target_label = "Target:"
+    target = ""
+
+    for line in lines.splitlines(True):
+        line = line.strip().split()
+        if line[0] == target_label:
+            target = line[1]
+
+    return target
+
+
+def dump_compiler_info(output_path, filename, data):
+    filename = os.path.join(output_path, filename)
+    all_data = dict()
+    if os.path.exists(filename):
+        with open(filename, 'r') as f:
+            all_data = json.load(f)
+    all_data.update(data)
```

Comment:
be careful, this will not add a new member in the hash, but update an existing one. Thus you will always have one element in the dict such as `{'target': 'arm', 'compiler': 'gcc-arm'}`

Reply:
Fixed in a new squashed commit